tests/phpunit/api/v3/LoggingTest.php

```diff
@@ -150,16 +156,20 @@ public function testUpdateLogTableHookINNODB() {
 
   /**
    * Check that if a field is added then the trigger is updated on refresh.
+   *
+   * @throws \CRM_Core_Exception
    */
-  public function testRebuildTriggerAfterSchemaChange() {
+  public function testRebuildTriggerAfterSchemaChange(): void {
     $this->callAPISuccess('Setting', 'create', ['logging' => TRUE]);
     $tables = ['civicrm_acl', 'civicrm_website'];
     foreach ($tables as $table) {
       CRM_Core_DAO::executeQuery("ALTER TABLE $table ADD column temp_col INT(10)");
     }
+    unset(\Civi::$statics['CRM_Logging_Schema']);
 
     $schema = new CRM_Logging_Schema();
-    $schema->fixSchemaDifferencesForAll(TRUE);
+    $schema->fixSchemaDifferencesForAll();
+    Civi::service('sql_triggers')->rebuild();
```

Comment:
I think the first run of the test failed because it needs `rebuild($tables=NULL, $force=TRUE)` to be equivalent to the previous call to `triggerRebuild($tables=NULL, $force=TRUE);` If that works, then +1 for merging.

Alternatively - you might be able to swap the two lines with just a call to `$schema->fixSchemaDifferences()` or `$schema->fixSchemaDifferences(TRUE)`, since that function basically does these two lines.

Reply:
Yeah - I'm trying to get rid of that $force being passed in because I don't think we should be offering it as a parameter purely to support this test - might take a couple of rounds fight with jenkins to get there

Reply:
Ok - unsetting that static worked